dask/dataframe/io/parquet/arrow.py

```diff
@@ -1636,7 +1637,8 @@ def _get_dataset_object(paths, fs, filters, dataset_kwargs):
         #       existence of _metadata.  Listing may be much more
         #       expensive in storage systems like S3.
         allpaths = fs.glob(paths[0] + fs.sep + "*")
-        base, fns = _analyze_paths(allpaths, fs)
+        allpaths = sorted(allpaths, key=natural_sort_key)
+        _, base, fns = _sort_and_analyze_paths(allpaths, fs)
```

Comment:
How is this different from `allpaths, base, fns = _sort_and_analyze_paths(allpaths, fs)` ?

Reply:
good catch, fixed